### PR TITLE
TYP: remove ignore in optional_args

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref #37715

### CODE REVIEW

File: pandas/_testing/_io.py [hunk 1]
```diff
@@ -82,10 +82,7 @@ def dec(f):
         is_decorating = not kwargs and len(args) == 1 and callable(args[0])
         if is_decorating:
             f = args[0]
-            # pandas\_testing.py:2331: error: Incompatible types in assignment
-            # (expression has type "List[<nothing>]", variable has type
-            # "Tuple[Any, ...]")
-            args = []  # type: ignore[assignment]
+            args = ()
```
Comment (jreback) [inline pandas/_testing/_io.py:85]: this should be able to just be removed entirely

Comment (simonjayhawkins) [reply pandas/_testing/_io.py:85]: hmm, we call `dec` with `f` and it uses the `args` from the outer scope. I assume that zeroing `args` out is intentional?

**Comment by jreback** [comment]
@simonjayhawkins status?

**Comment by simonjayhawkins** [comment]
closing while we discuss #39513

**Pull request closed without merging**